flatnav/util/VisitedNodesHandler.h

```diff
@@ -0,0 +1,175 @@
+#pragma once
+
+#include <flatnav/util/SIMDDistanceSpecializations.h>
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <stdint.h>
+#include <vector>
+
+namespace flatnav {
+
+class VisitedNodesHandler {
+private:
+  uint32_t _mark;
```

Comment:
Does this need to be a unit32? The internal array is same size as N (number of nodes) - at billion scale, each VisitedNodesHandler would be 4 GB.

Reply:
I think that would still fit into a `uint32_t` since it can hold up to N = 2^32 - 1 = 4,294,967,295.

Reply:
No, my question was whether we could have it be a char (as this would reduce the per-thread overhead from 4 GB to 1 GB)

Reply:
yeah, you're right. `unsigned char` aka `uint8_t` is more suitable here.